### PR TITLE
Add error handling to zip code and style results page better

### LINKED ISSUE #20: style results page

### PR DESCRIPTION
**Pull request by clairefields15**
### Acceptance Criteria
Added more styling to the results cards, they look decent now but could probably look better.

####Fixed bug in Search/added more error handling
bugs: press enter to search didn't work, was directing to http://localhost:3000/?zip=32301
error handling: there was no way to disable the button if there wasn't a zip there

expected behavior:
1- enter a zip code in the field and if you press the button see localhost:3000/markets and results for it
2 - enter a zip code in the field and press enter and see localhost:3000/markets and results for it

Fixed the bug by changing the Link to a button (which allowed me to disable the button if there wasn't a valid zip.. fun fact, Links can't be disabled), routing now happens in App on lines 52/53.  Paired with my mentor for this solution using history.push, using Redirect instead would have been tricky.

### Issues Resolved
closes #20 

### Code Reviewers
@alexmkio @AshleyOh-bit @Meekb

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -48,7 +49,8 @@ export const App: React.FC = () => {
       let data = await response.json();
       let cleanedData = cleanMarketsData(data.results);
       setMarkets(cleanedData);
-      getDetails(cleanedData);
+      await getDetails(cleanedData);
```
Comment (alexmkio) [inline src/App.tsx:52]: This function call isn't async. Is await needed here?

[I believe this is only needed for the resolution of Promises which is happening in the fetch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -8,16 +9,22 @@ interface getMarkets {
...
-  const submitZipSearch = () => {
-    zip = zip.replace(/[^0-9]/g, '');
-    zip = zip.substring(0, 5);
-    setZip(zip);
-    getMarkets(zip);
-    clearInput();
-  };
+  useEffect(() => {
+    const validZip = new RegExp('^[0-9]{5}(?:-[0-9]{4})?$');
+    if (validZip.test(zip)) {
```
Comment (alexmkio) [inline src/components/Search/Search.tsx:16]: whoa! Nice!!!

File: src/App.tsx (hunk 1 above)
Comment (clairefields15) [reply src/App.tsx:52]: It's added so that you wait for details to be in state before routing to the results url

Comment (alexmkio) [reply src/App.tsx:52]: I just tried it without await and it has no effect.
Screen Shot 2021-07-28 at 9 06 41 PM

Comment (Meekb) [reply src/App.tsx:52]: Would love to run through the async / await process when we meet up tomorrow  to make sure I fully understand its purpose

**Review by Meekb — APPROVED** [review]
Nice work!

File: src/App.tsx (hunk 1 above)
Comment (clairefields15) [reply src/App.tsx:52]: I'd like to try it with throttled internet and see what the ux difference is since my mentor suggested I add it

**Review by alexmkio — APPROVED** [review]

**Pull request merged**